packages/react/src/utilities/types.ts

```diff
@@ -0,0 +1,5 @@
+/** Simplify a type signature to help with readability in editor hints, error messages etc */
+export type Simplify<T> = { [Key in keyof T]: T[Key] } & {};
+
+/** Merge two types, with properties from the second overriding properties from the first */
+export type Merge<T1, T2> = Simplify<Omit<T1, keyof T2> & T2>;
```

Comment:
I guess this is to make it easier overriding types when using intersections without having to omit the type? 🤔 

Suggest calling this `mergeRight` as is the common naming convention for these kinds of operation, in functional programming atleast 😅